// @vonage/voice/dist/esm/ncco.mjs
import {
  Input
} from "./chunk-FUFZV4BW.mjs";
import {
  NCCOBuilder
} from "./chunk-TRQVMZID.mjs";
import {
  Notify
} from "./chunk-IAXFZXU4.mjs";
import {
  Record
} from "./chunk-DUB4CYRQ.mjs";
import {
  Stream
} from "./chunk-4F5264S3.mjs";
import {
  Talk
} from "./chunk-UMZUQWAH.mjs";
import "./chunk-PAMOQNSI.mjs";
import "./chunk-SOEE7WVN.mjs";
import "./chunk-VGQVK3CR.mjs";
import "./chunk-TCQZSCGM.mjs";
import {
  Connect
} from "./chunk-HEJF6T7A.mjs";
import {
  Conversation
} from "./chunk-CWHH465G.mjs";
import "./chunk-FKHL5TYE.mjs";
import "./chunk-2WSLQJS2.mjs";
import "./chunk-3NG6QREM.mjs";
import "./chunk-CDEWI66T.mjs";
import "./chunk-XNAXG2TG.mjs";
import "./chunk-E6Z2ASAV.mjs";
import "./chunk-C5TVTVBY.mjs";
import "./chunk-GIT4DK2J.mjs";
import "./chunk-V66WSPX3.mjs";
import "./chunk-RNTVM7DX.mjs";
import "./chunk-3PSD3IGD.mjs";
import "./chunk-ETZTGOV6.mjs";
import "./chunk-GMDYX2A5.mjs";
import "./chunk-RFGO5BC2.mjs";
import "./chunk-IXODLYK5.mjs";
import "./chunk-Y3T6C6QB.mjs";
import "./chunk-FNFQ27I4.mjs";
import "./chunk-NSLUVWNN.mjs";
import "./chunk-NVSQ62IY.mjs";
import "./chunk-6UJQTCSN.mjs";
import "./chunk-QE7UENVE.mjs";
import "./chunk-3FQRF2AX.mjs";
import "./chunk-BHJ6KK3Z.mjs";

// lib/ncco.ts
import debug from "debug";
debug("@vonage/voice")(
  "This file is deprecated please update to use the core package"
);
export {
  Connect,
  Conversation,
  Input,
  NCCOBuilder,
  Notify,
  Record,
  Stream,
  Talk
};
